### ISSUE TITLE
Invisible rectangle, potentially from data table, violates G18 (contrast ratio against the background)

### ISSUE THREAD

**Issue by jandraa** [carbon-design-system/carbon#6815]
## Brief description

An invisible rectangle causes failure on the A11y checker saying, the contrast ratio of text with its background must meet WCAG 2.1 AA requirements. This happens in at least 5 different instances when a modal is triggered from a data table, like in the case of an edit modal.

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?

Nothing. The contrast ratio of text with its background (i.e. background with a color gradient or a background image) must meet WCAG 2.1 AA requirements

> What happened instead?

A11y Checker tool reported 2 violations this rule for following element:

    <span class="bx--visually-hidden" role="link" tabindex="0">

> What WCAG 2.1 checkpoint does the issue violate?

[G18](https://www.w3.org/WAI/WCAG21/Techniques/general/G18) Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text.

## Steps to reproduce the issue

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information

- Screenshots or code
![image](https://user-images.githubusercontent.com/44675611/92742105-ec89c900-f344-11ea-9a8c-3c244ade37fd.png)
![image](https://user-images.githubusercontent.com/44675611/92743518-3b842e00-f346-11ea-9924-29c8bf3b147d.png)
![image](https://user-images.githubusercontent.com/44675611/92743527-417a0f00-f346-11ea-9925-4aa54448690a.png)
![image](https://user-images.githubusercontent.com/44675611/92743541-450d9600-f346-11ea-8179-483059b4ee72.png)
![image](https://user-images.githubusercontent.com/44675611/92743563-4b9c0d80-f346-11ea-9281-faa46cbb87e1.png)

**Comment by tw15egan** [comment]
@dakahn I believe this is coming from the `focus sentinel` used to handle focus trapping in the Modal component

https://github.com/carbon-design-system/carbon/issues/5545

**Comment by dakahn** [comment]
Interesting this is cropping back up again. We can forward this report to the AC team for resolution/next steps

**Comment by dhyliu** [comment]
@joshblack Josh, I noted the referenced modal issue has been closed. This issue has been opening since Sep, 2020. May I have an update to the status of this issue?

**Comment by dakahn** [comment]
can't reproduce with our React Storybook. @jandraa is this issue still present while using latest Carbon React? If so can you recreate in a Codesandbox so we can verify? Thanks 👍🏽

**Comment by PaigeChen** [comment]
Hi @dakahn , we're using v7.27 (released in late Jan) and still can reproduce this issue. It's my very first time trying to recreate the issue on Codesandbox. Would you mind to provide some guidance? Thanks.

**Comment by joshblack** [comment]
Hi @PaigeChen! Here is the link to our codesandbox template that you can use: https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

In it, you can pull components from Carbon to use and from that steps to reproduce would be perfect

**Comment by PaigeChen** [comment]
Hi @joshblack , thanks for the introductions. I started a sandbox and adjusted the carbon component versions to the ones we use. After checking the ones others created, I guess the relevant codes that will generate the Modal component should be put in it? 
Sorry if I'm asking newbie questions, just want to make sure I'm creating the test env correctly. Thanks.

**Comment by PaigeChen** [comment]
Asked around and got support from my co-worker for how to re-build a test env on codesandbox properly. However, I cannot reproduce it. Below are the carbon versions used in the sandbox env. Any advise about what I can try to reproduce it? Thanks.

carbon-components : 10.27.0	
carbon-components-react : 7.27.0
carbon-icons : 7.0.7 (latest)

**Comment by tw15egan** [comment]
If it is not able to be reproduced in a reduced test case, then it does not seem like it is an issue with the Carbon component itself but rather the implementation in the app. Do you have a link to the CodeSandbox? 

Is it possible you are using an out-of-date ruleset? 

![Screen Shot 2021-03-12 at 11 01 09 AM](https://user-images.githubusercontent.com/11928039/110986221-42eeb900-8322-11eb-858b-cea0caf9c42d.png)


You can see the offending element, the Modal focus sentinel is present in the storybook but doesn't seem to be throwing any issues.

![Screen Shot 2021-03-12 at 10 59 03 AM](https://user-images.githubusercontent.com/11928039/110986055-0de26680-8322-11eb-90cc-35e3d45bcc7a.png)

![Screen Shot 2021-03-12 at 11 00 34 AM](https://user-images.githubusercontent.com/11928039/110986202-379b8d80-8322-11eb-80e8-cf4f2db4de78.png)

**Comment by PaigeChen** [comment]
Hi @tw15egan , thanks for the reply. Yes, I'm sure that I'm using the latest ruleset(same as your screenshot) in A11y checker. And `<span tabindex="0" role="link" class="bx--visually-hidden">Focus sentinel</span>` appears in my application's HTML as well.

The codesandbox link is https://codesandbox.io/s/black-snow-55jf9. Thanks.

**Comment by tw15egan** [comment]
Hmm well since this doesn't seem to be reproducible, I'm not sure what we can do on our end. I'm going to close this, but if you figure out a way to get the violation to show I'll gladly reopen and take a look. Thanks!

**Comment by PaigeChen** [comment]
Hi TJ, thanks a lot for the comments. We'll keep trying to get it work on sandbox and update this issue if we have any progress. Thanks for your support!
